articles/libraries/lock-android/v1/use-your-own-ui.md

```diff
@@ -0,0 +1,275 @@
+          this.profile = profile;
+          this.token = token;
+      }
+
+      public Token getToken() { return token; }
+
+      public UserProfile getProfile() { return profile; }
+  }
+
+  public class ErrorEvent {
+      private Dialog dialog;
+      private int title;
+      private int message;
+      private Throwable throwable;
+
+      public ErrorEvent(Dialog dialog) {
+          this.dialog = dialog;
+      }
+
+      public ErrorEvent(int title, int message, Throwable throwable) {
+          this.title = title;
+          this.message = message;
+          this.throwable = throwable;
+      }
+  }
+  ```
+
+Then in your Login *Activity* add the following fields
+  ```java
+      private AuthenticationAPIClient client;
+      private EventBus eventBus;
+  ```
+
+In the same *Activity* `onCreate` method add the following lines to initialize **Auth0** and the fields we added earlier
+  ```java
+      Auth0 auth0 = new Auth0(getString(R.string.auth0_client_id), getString(R.string.auth0_domain_name));
+      this.client = auth0.newAuthenticationAPIClient();
+      this.eventBus = new EventBus();
+  ```
+
+When you need to login your user with email/password credentials, just paste the following code
+  ```java
+      String email = ... // Get email
+      String password = ... // Get password
+      //add your DB connection name, since you are using email / pwd login
+      //will throw a 400 error if not set
+      ParameterBuilder parameterBuilder = ParameterBuilder.newBuilder();
+      Map<String, Object> authenticationParameters = parameterBuilder
+                .setConnection("YOUR_DB_CONNECTION_NAME")
+                .asDictionary();
+      client.login(email, password)
+          .addParameters(authenticationParameters)
+          .start(new AuthenticationCallback() {
+              @Override
+              public void onSuccess(UserProfile userProfile, Token token) {
+                  eventBus.post(new AuthenticationEvent(userProfile, token));
+              }
+
+              @Override
+              public void onFailure(Throwable throwable) {
+                  eventBus.post(new ErrorEvent(R.string.login_failed_title, R.string.login_failed_message, throwable));
+              }
+          });
+  ```
+  > For more details about the parameters you can check [this wiki page](/libraries/lock-android/sending-authentication-parameters).
+
+Finally handle both `AuthenticationEvent` and `ErrorEvent`
+  ```java
+      @Override
+      protected void onStart() {
+          super.onStart();
+          eventBus.register(this);
+      }
+
+      @Override
+      protected void onStop() {
+          super.onStop();
+          eventBus.unregister(this);
+      }
+
+      public void onEvent(ErrorEvent event) {
+        //Handle Error
+      }
+
+      public void onEvent(AuthenticationEvent event) {
+        //Handle authentication
+      }
+  ```
+
+## Social Authentication
+
+Include the following libraries in your `build.gradle`:
```

Comment:
Add the "native provider" warning that I told you above, as this seems to be using the native libraries.